### PR TITLE
t0404-currentColor: Make css extraction respect ordering

### PR DESCRIPTION
**Pull request by Implausiblyfun**
This is a gnarly bit of rewrites in the css.go piece.
In this PR we have done the following:
1. Keep track of selector order during the css.go loop. This means that we have a last point in the css file where order still exists which allows for deterministic overwriting of values. This is done for emphasis on some css test files but most realistically seems to be useful when there is a higher order combined directive that is overwritten by a more specific one. Such as border followed by a border-bottom-color or something like that.
2. Add mechanism to unroll combined directives such as "border" at the css level rather than in our html as we still have ordering and can perform the right overwriting.
3. Add an true by default (rather than the correct false by default) validation step for setting things like "color". 

In addition to these major changes we also now support inherited color "currentcolor" in more places like borders which previously ignored this value.

### CODE REVIEW

File: testdata/htmlout/t040205-hsla-basic-a.png
Comment (Implausiblyfun) [inline testdata/htmlout/t040205-hsla-basic-a.png:1]: This is due to adding margin splitting via the compositeAttributeMapping table
ie margin is meant to be 0 on this one but we werent respecting it before

File: css.go [hunk 1]
```diff
@@ -427,3 +444,124 @@ func ParseAttributeSelector(selector string) (AttributeSelector, error) {
 		matchFn: match,
 	}, nil
 }
+
+type procesor func(map[string]string, string, string)
```
Comment (200sc) [inline css.go]: nit: procesor -> processor

Comment (Implausiblyfun) [reply css.go]: nice call. I wonder if its worth while to https://github.com/marketplace/actions/github-spellcheck-action or something as a ci action?

Comment (200sc) [reply css.go]: Yeah golangci-lint + that (unless golangci-lint has typos built in) would be good

**Review by 200sc — APPROVED** [review]

**Pull request merged**